Services/Mentor/OJS.Services.Mentor.Business/Implementations/MentorBusinessService.cs

```diff
@@ -0,0 +1,480 @@
+        {
+            if (element is Paragraph paragraph)
+            {
+                var styleId = paragraph.ParagraphProperties?.ParagraphStyleId?.Val?.Value;
+                var text = paragraph.InnerText.Trim();
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    continue;
+                }
+
+                var isHeading = !string.IsNullOrEmpty(styleId) && (styleId.StartsWith("Heading1", StringComparison.Ordinal) || styleId.StartsWith("Heading2", StringComparison.Ordinal));
+
+                if (isHeading)
+                {
+                    // Case 1: Check if this heading matches the problem name
+                    if (problemName.Contains(text, StringComparison.OrdinalIgnoreCase))
+                    {
+                        currentHeading = text;
+                        shouldCollectContent = true;
+                        resultContent.Clear(); // Clear any previous content
+                        continue;
+                    }
+
+                    // If we were collecting content for a matching heading, we're done
+                    if (shouldCollectContent && currentHeading != null && sectionCount < problemNumber)
+                    {
+                        return resultContent.ToString().Trim();
+                    }
+
+                    sectionCount++;
+
+                    // Case 2: Check if we should start collecting from this section based on problem number
+                    if (sectionCount == problemNumber)
+                    {
+                        shouldCollectContent = true;
+                        resultContent.AppendLine($"## {text}");
+                    }
+                    else if (sectionCount > problemNumber)
+                    {
+                        resultContent.AppendLine($"## {text}");
+                    }
+
+                    currentHeading = text;
+                }
+                else
+                {
+                    // For Case 3, collect everything
+                    if (problemNumber > sectionCount)
+                    {
+                        resultContent.AppendLine(text);
+                    }
+                    // For other cases, collect only when flagged
+                    else if (shouldCollectContent)
+                    {
+                        resultContent.AppendLine(text);
+                    }
+                }
+            }
+            else if (element is Table table)
+            {
+                // For Case 3, collect all tables
+                if (problemNumber > sectionCount)
+                {
+                    ProcessTable(table, resultContent);
+                }
+                // For other cases, collect only when flagged
+                else if (shouldCollectContent)
+                {
+                    ProcessTable(table, resultContent);
+                }
+            }
+        }
+
+        return resultContent.ToString().Trim();
```

Comment:
Is there a way for this to return empty string? We should handle it by adding the problem and contest name and instruct the bot to look for it in it's own data.

Reply:
Done.